### ISSUE TITLE
upcoming sf 1.0-0 breaks himach on CRAN

### ISSUE THREAD

**Issue by edzer** [david6marsh/himach#8]
Please see https://github.com/r-spatial/sf/issues/1649

As you are aware sf 1.0-0 uses spherical geometry when working with ellipsoidal coordinates. Your package turns this off, but `sf` now prints a message to the user to notify it has been switched off. We see:
```
    `himach:::.onLoad()` produced output.
```
indicating that you should do this in `.onAttach()` rather than in `.onLoad`.

**Comment by david6marsh** [comment]
Thanks, I'll check at
the weekend when I'm back at my desk.

I'm surprised you say it turns it off, I'm fairly sure himach insists on s2
by default. I was so happy when you did the s2 implementation; it solved a
lot of challenges I was facing. So thanks a lot for that work.

David

On Wednesday, June 9, 2021, Edzer Pebesma ***@***.***> wrote:
> Please see r-spatial/sf#1649
>
> As you are aware sf 1.0-0 uses spherical geometry when working with
ellipsoidal coordinates. Your package turns this off, but sf now prints a
message to the user to notify it has been switched off. We see:
>
> `himach:::.onLoad()` produced output.
>
> indicating that you should do this in .onAttach() rather than in .onLoad.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.<
https://ci4.googleusercontent.com/proxy/KyZoWRJVXINhgB-UmQcLAnuPNdTSA3n2HHBOzQOSgHooo0AEi_rKjrpY4NXDwfJ80-XxMSXmR6erCoREe31ABzltzZ6AoNYdsSS9c8uMIbljo6wLT2_jO1fmqcOIgD_b2w4kB1lkdOBy2kyutGQ8BLJlsUz13hAZhYNmf8Mtl36GGhwDOPeLuGvHpY3WFaIe3k49CKfBP-y0RLVD_78aBX4q784=s0-d-e1-ft#https://github.com/notifications/beacon/AKLAZBWDM7HC2TRZCKOAMOTTR5REHA5CNFSM46L77VO2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4NU3YQ3A.gif>

**Comment by edzer** [comment]
Yes, I misinterpreted the error: the package seems fine, it's `test-zzz.R` that causes the problems with:
```
  sf::sf_use_s2(FALSE)
  expect_silent(himach:::.onLoad())
```
because `sf::sf_use_s2(TRUE)` _after_  `sf::sf_use_s2(FALSE)` causes output.

**Comment by david6marsh** [comment]
I've pushed a patch to GitHub (v0.2.1) that silently ignores the new messages. CMD check ok with the latest version of `sf` (v1.0-1) just downloaded from GitHub, also ok with sf v1.0-0 from CRAN.

Do you need `himach` v0.2.1 pushed immediately to CRAN? If not, I'd rather wait.

**Comment by edzer** [comment]
I don't need anything from you, but CRAN will archive packages that have ERROR results [here](https://cran.r-project.org/web/checks/check_results_himach.html) after a while: they expect you to address them.

**Comment by david6marsh** [comment]
Fixed in v0.2.1
